### PR TITLE
UCT/CUDA/UCM: Add support for cuda unified memory

### PR DESCRIPTION
**Pull request by bureddy**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6766/ for details (Mellanox internal link).

File: src/ucm/event/event.c [hunk 1]
```diff
@@ -554,6 +571,24 @@ cudaError_t ucm_cudaMalloc(void **devPtr, size_t size)
...
 
+cudaError_t ucm_cudaMallocManaged(void **devPtr, size_t size, unsigned int flags)
+{
+    cudaError_t ret;
+
+    ucm_event_enter();
+
+    ret = ucm_orig_cudaMallocManaged(devPtr, size, flags);
+    if (ret == cudaSuccess) {
+        ucm_trace("ucm_cudaMallocManaged(devPtr=%p size:%lu flags;%d)",
```
Comment (brminich) [inline src/ucm/event/event.c]: minor: need to use `:` instead of `;`

File: src/ucp/core/ucp_context.c [hunk 2]
```diff
@@ -46,7 +46,8 @@ static const char * ucp_rndv_modes[] = {
 };
 
 uct_memory_type_t ucm_to_uct_mem_type_map[] = {
-    [UCM_MEM_TYPE_CUDA] = UCT_MD_MEM_TYPE_CUDA,
+    [UCM_MEM_TYPE_CUDA]         = UCT_MD_MEM_TYPE_CUDA,
+    [UCM_MEM_TYPE_CUDA_MANAGED] = UCT_MD_MEM_TYPE_HOST,
```
Comment (brminich) [inline src/ucp/core/ucp_context.c]: no need for `,`

File: test/gtest/ucm/cuda_hooks.cc [hunk 3]
```diff
@@ -92,16 +92,16 @@ class cuda_hooks : public ucs::test {
     }
 
 
-    void check_mem_alloc_events(void *ptr, size_t size) {
+    void check_mem_alloc_events(void *ptr, size_t size, int expect_mem_type) {
```
Comment (brminich) [inline test/gtest/ucm/cuda_hooks.cc]: minor: maybe declare default value for `expect_mem_type` as `UCM_MEM_TYPE_CUDA`. Then you would not need to add it in the old tests.

Comment (bureddy) [reply test/gtest/ucm/cuda_hooks.cc]: fixed

File: src/ucm/event/event.c (hunk 1 above)
Comment (bureddy) [reply src/ucm/event/event.c]: fixed

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6777/ for details (Mellanox internal link).

**Pull request merged**